tests/scripts/configs_e2e/configs_e2e.go

```diff
@@ -0,0 +1,408 @@
+	for _, f := range filesToCopy {
+		if err := copyFile(path.Join(outputDir, f), path.Join(srcDir, f)); err != nil {
+			return fmt.Errorf("error copying %q from %q to %q: %v", f, srcDir, outputDir, err)
+		}
+		log.Printf("Copied %q from %q to %q.", f, srcDir, outputDir)
+	}
+	if err := createWorkspaceFile(m, configTarballURL, outputDir); err != nil {
+		return fmt.Errorf("error creating the Bazel WORKSPACE file: %w", err)
+	}
+	if err := createBazelrcFile(m, configTarballURL, outputDir, rbeInst); err != nil {
+		return fmt.Errorf("error creating the .bazelrc file: %w", err)
+	}
+	return nil
+}
+
+func validateRBEInstName(instName string) error {
+	wantFormat := "projects/<GCP project ID>/instances/<instance ID>"
+	splitName := strings.Split(instName, "/")
+	if len(splitName) != 4 {
+		return fmt.Errorf("%q did not conform to format %q because it split into %d elements by '/' instead of 4", instName, wantFormat, len(splitName))
+	}
+	if splitName[0] != "projects" {
+		return fmt.Errorf("%q did not conform to format %q because the first element was %q instead of 'projects'", instName, wantFormat, splitName[0])
+	}
+	if splitName[2] != "instances" {
+		return fmt.Errorf("%q did not conform to format %q because the third element was %q instead of 'instances'", instName, wantFormat, splitName[2])
+	}
+
+	return nil
+}
+
+// downloadBazelisk downloads Bazelisk for Linux to the given directory and returns the path to the
+// downloaded Bazelisk executable.
+func downloadBazelisk(outputDir string) (string, error) {
+	bazeliskURL, bazeliskFile, err := rbeconfigsgen.BazeliskDownloadInfo(rbeconfigsgen.OSLinux)
+	if err != nil {
+		return "", fmt.Errorf("unable to determine URL to download Bazelisk from for Linux: %w", err)
+	}
+	resp, err := http.Get(bazeliskURL)
+	if err != nil {
+		return "", fmt.Errorf("unable to initialize the Bazelisk download from %q: %w", bazeliskURL, err)
+	}
+	defer resp.Body.Close()
+
+	bazeliskPath := path.Join(outputDir, bazeliskFile)
+	o, err := os.Create(bazeliskPath)
+	defer o.Close()
+
+	log.Printf("Downloading Bazelisk from %s to %s.", bazeliskURL, bazeliskPath)
+	if _, err := io.Copy(o, resp.Body); err != nil {
+		return "", fmt.Errorf("error while downloading Bazelisk from %q to %q: %w", bazeliskURL, bazeliskPath, err)
+	}
+
+	return bazeliskPath, nil
+
+}
+
+// runTestBuild runs the remote build using the toolchain configs using Bazelisk to pin the version
+// of Bazel.
+func runTestBuild(ctx context.Context, workingDir, bazelVersion string) error {
+	bazeliskPath, err := downloadBazelisk(workingDir)
+	if err != nil {
+		return fmt.Errorf("failed to download Bazelisk: %w", err)
+	}
+	if err := os.Chmod(bazeliskPath, os.ModePerm); err != nil {
+		return fmt.Errorf("unable to update the permissions of downloaded Bazelisk binary %q to make it executable: %w", bazeliskPath, err)
+	}
+	ctxWithDeadline, cancel := context.WithTimeout(ctx, 3*time.Minute)
```

Comment:
Maybe pull the `3*time.Minute` bit out to a constant, or command line flag-backed var?

Reply:
Done. Moved to flag